site/content/entry/citra-progress-report-2017-august.md

```diff
@@ -0,0 +1,155 @@
+malicious game could just ask Citra to change `/../../../../Documents/IMPORTANT.docx`'s
+name to `/delicious_secrets.docx`, effectively moving `IMPORTANT.docx` into Citra's
+virtual SD card! And from there, the game could just ask Citra to open
+`/delicious_secrets.docx` to read the file it was originally forbidden from accessing.
+
+This patch now fixes this, such that if any 3DS game or application tries to do
+this from inside Citra, rather than doing what it asks, Citra simply crashes
+itself, as it doing that could mean the game had intentions to do something evil.
+So far, we haven't found any real 3DS software that tries to do this, but at
+least now future attempts to do so are blocked.
+
+## [Fix Edge Cases for TextureCopy](https://github.com/citra-emu/citra/pull/2809) by [wwylele](https://github.com/wwylele)
+
+The 3DS GPU has a data transfer mode called `TextureCopy`, which as the name says,
+is for copying textures but with a configurable gap in case the texture is going
+to be copied into a smaller resolution area. This mode is enabled by setting the
+3rd flag in the GPU and causes it to ignore every other flag except the 2nd, which
+is used to tell it whether or not to crop the texture. 
+
+This is nice because it can be used as a quick and easy way for 3DS developers to
+duplicate textures, and can be used in situations such as the one pictured running
+on hardware in Pokémon Super Mystery Dungeon below. This is really nice for those
+creating games, but when it came to running something that took advantage of this
+feature in Citra it didn't always work the same as it would on console. This
+provided us with an accuracy issue that needed to be solved.
+
+{{< figure src="/images/entry/citra-progress-report-2017-august/texturecopy-before.png" 
+    title="How jagged" alt="Pokémon Super Mystery Dungeon During Deoxy's and Rayquaza's Face Off   I N   S P A C E" >}}
+
+Fortunately (and to much rejoice) wwylele stepped into the ring to wrestle with
+this issue. They prepared a [test program](https://github.com/wwylele/ctrhwtest/tree/master/texture-copy-test)
+to help gain an understanding of how hardware handles the TextureCopy operation
+in comparison to Citra.
+
+{{< figure src="/images/entry/citra-progress-report-2017-august/texturecopy-after.png" 
+    title="Deoxys is having a bit of a hard time, no?" alt="Pokémon Super Mystery Dungeon During Deoxy's and Rayquaza's Face Off   I N   S P A C E" >}}
+
+## [Use Docker For Linux Builds](https://github.com/citra-emu/citra/pull/2869) by [j-selby](https://github.com/j-selby)
+
+Ubuntu Linux 14.04 is the de-facto standard desktop Linux distribution. It's also
+[old](https://wiki.ubuntu.com/TrustyTahr/ReleaseSchedule#line-37), *very* old.
+<span style="font-size:7pt; vertical-align:super;">
+(It's the Windows XP of Linux, really. :P)</span> So old, in fact, that the
+standard library it ships with can't compile Citra. And our buildbot,
```

Comment:
Feel like this could be worded a little better (my fault in explaining) - we pulled a really new version of gcc/g++ from a 3rd party repo that also, by consequence, pulled in a incompatible `libc++`. We need this newer g++ as older versions don't support Citra. Travis *itself* doesn't ship with this repository, but by default, it cannot compile Citra.